cusub_cortex/state_machine/src/tasks/search.py

```diff
@@ -74,32 +78,28 @@ def query_listener(self):
             for key in dobj_dict.keys():
                 dobj_nums = dobj_dict[key]
                 for num in dobj_nums:
-                    if self.det_listener.check_new_dv(num):
+                    if self.det_listener.check_new_dv(num, clear_flag=False):
                         return True
         return False
+    
+    def clear_new_flags_dvs(self):
+        dobj_dict = self.det_listener.query_classes(self.target_classes)
+        if not dobj_dict: # No classes present
+            return False
+        else:
```

Comment:
This seems like a useful function for any Objective. Should this be in the Objective class to allow inheritance?

Reply:
yeah perhaps a version of this function could replace the big block at the beginning of an execute function in several tasks that just figures out which dobject corresponds to the target class